src/frontend/sharedComponents/HookFormTextInput.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import {
+  Control,
+  Controller,
+  FieldValues,
+  Path,
+  useWatch,
+  ValidationRule,
+  useFormState,
+  UseControllerProps,
+} from 'react-hook-form';
+import {TextInput as RNTextInput, StyleSheet, View} from 'react-native';
+import {BLACK, LIGHT_GREY, RED} from '../lib/styles';
+import {ErrorIcon} from './icons';
+import {ViewStyleProp} from '../sharedTypes';
+import {Text} from './Text';
+
+type TextInputProps<InputFields extends FieldValues> = {
+  containerStyle?: ViewStyleProp;
+  showCharacterCount?: boolean;
+  control: Control<InputFields>;
+} & Omit<
```

Comment:
just a note that `Omit` is rather expensive in Typescript (especially on object types) so ideally we don't need this unless you're actually providing specific native text input props when using this component

Reply:
saw your other comment about why this is included. would consider the suggestion in https://github.com/digidem/CoMapeo-mobile/pull/156#discussion_r1458028985

Reply:
so were only using `name` and `control`. So do you think we should just use `Pick< ... ,'name'>` and not expose the other ones?

Reply:
Ohhh nevermind this comment is referring to textInput props, not Controller Props

Reply:
https://github.com/digidem/CoMapeo-mobile/pull/156/commits/172927279e7822ccb2b3f4496ceafa430ed5b5e6

Reply:
referring to wrong commit? not sure what i should be looking at 😅

Reply:
yup, sorry you can ignore